### PR TITLE
Fix copying cargo artifacts from deps

### LINKED ISSUE #202: Cargo artifacts not being copied from deps derivation
I noticed that building my crate via naersk was resulting in cargo rebuilding the entire dependency tree from scratch every time, even when the Cargo.lock had not changed.

Upon closer inspection I discovered that the deps derivation results in a `target.tar.zstd` which unarchives to an empty directory.

I think the culprit may be [forcing cargo to use `out` as its target directory](https://github.com/nix-community/naersk/blob/074d81b1a45145f076b2adf93184073fc9615397/build.nix#L243) but then [copying the artifacts from the (unused) `target` directory](https://github.com/nix-community/naersk/blob/074d81b1a45145f076b2adf93184073fc9615397/build.nix#L402-L415).

### PR DESCRIPTION
**Pull request by ipetkov**
* Previously the CARGO_TARGET_DIR environment was set to the `out`
  directory which resulted in cargo placing all of its artifacts there
  instead of the default `target` directory
* Later on, the build scripts would then copy the (empty) `target`
  directory to `$out` resulting in cargo rebuilding the entire
  dependency tree from scratch whenever the top level crate's source was
  changed
* Now we omit the CARGO_TARGET_DIR environment variable and inspect the
  `target` directory for any binaries instead of the `out` directory

Fixes #202

### CODE REVIEW

**Comment by nmattia** [comment]
Thanks for the PR! I fixed part of the CI in #204 . There's one issue remaining: #205

Re. your PR, how does it relate to https://github.com/nix-community/naersk/pull/201?

**Comment by ipetkov** [comment]
> Re. your PR, how does it relate to #201?

Ah I hadn't noticed it before submitting this PR. I think they both solve different issues in opposite ways. As blitz commented on the other #201, it shouldn't be needed if we merge #203

IMO #203 is a more general solution, as the next person to come along and make a change can expect cargo artifacts to be in `target` instead of needing to remember to use `out` everywhere (while also avoiding getting confused with `$out`)

**Review by nmattia — APPROVED** [review]
Mostly rubbing stamping here, there's been so many changes around how we copy the executables that I can't keep track 😂 

@blitz I closed #201 in favor of this, would you mind having a look and approving if it looks good?

File: test.nix
Comment (blitz) [inline test.nix:100]: nit: It would help if there is a small comment what these tests actually test for.

**Review by blitz — APPROVED** [review]
This looks good! It also fixes `doDoc = true`.

I've run the test suite manually and it reports success.

File: test.nix
Comment (ipetkov) [reply test.nix:105]: Done!

**Review by blitz — APPROVED** [review]
Thanks for the explanation in the tests. Much easier to understand now.

**Comment by blitz** [comment]
@nmattia Ping. This is ready to be merged. :)

**Comment by blitz** [comment]
@anderssorby can you take a look?

**Comment by Anderssorby** [comment]
@blitz Looks good to me. This is a much needed fix.

**Comment by blitz** [comment]
@nmattia has to merge this I guess. At least I can't.

**Comment by ipetkov** [comment]
Might be good to update the CODEOWNERS file with the new maintainers so we don't have to keep spamming nmattia :)

(If someone with write access wants to merge the commit and push directly that's cool too)

**Comment by blitz** [comment]
> Might be good to update the CODEOWNERS file with the new maintainers so we don't have to keep spamming nmattia :)

See #208.

**Comment by nmattia** [comment]
I'll let you do the honours :)

**Pull request merged**
